scripts/importer/data_importer/dbsnp_importer.py

```diff
@@ -0,0 +1,126 @@
+            self.in_file = self._open( os.path.join(self.download_dir, filename) )
+        except FileNotFoundError:
+            self.in_file = self._download_and_open(url)
+
+    def prepare_version(self):
+        version_id = "{a.dbsnp_version}_{a.dbsnp_reference}".format(a=self.settings)
+        if self.settings.dry_run:
+            try:
+                dbsnp_version = db.DbSNP_version.get(version_id = version_id)
+                logging.info("dbsnp_version '{}' already in database".format(version_id))
+            except db.DbSNP_version.DoesNotExist:
+                dbsnp_version = db.DbSNP_version.select(fn.Max(db.DbSNP_version.version_id)).get()
+                logging.info("Created dbsnp_version '{}'".format(version_id))
+        else:
+            dbsnp_version, created = db.DbSNP_version.get_or_create(version_id = version_id)
+            if created:
+                logging.info("Created dbsnp_version '{}'".format(version_id))
+            else:
+                logging.info("dbsnp_version '{}' already in database".format(version_id))
+        return dbsnp_version
+
+    def start_import(self):
+        """
+        dbsnp-file header is 'rsid', 'chrom', 'position'
+        """
+        dbsnp_version = self.prepare_version()
+
+        start = time.time()
+        last_progress = 0.0
+        logging.info("Inserting dbSNP data into database.")
+
+        counter = 0
+        batch = []
+        with db.database.atomic():
+            for line in self.in_file:
+                line = line.decode('ascii').strip()
+                if line.startswith("#"):
+                    continue
+
+                try:
+                    rsid, chrom, position = line.split("\t")[:3]
+                    position = int(position)
+                    position += 1 # 0-indexed
+                except ValueError:
+                    # we don't care for incomplete entries
+                    continue
+
+                if self.chrom and not chrom == self.chrom:
+                    continue
+
+                batch += [{ 'version':dbsnp_version,
+                            'rsid':rsid,
+                            'chrom':chrom,
+                            'pos':position}]
+                counter += 1
+
+                if self.total != None:
+                    progress = counter / self.total
+                    while progress > last_progress + 0.01:
+                        if not last_progress:
+                            logging.info("Estimated time to completion: {}".format(self._time_to(start, progress)))
+                            if self.total != None:
+                                self._print_progress_bar()
+                        self._tick()
+                        last_progress += 0.01
+
+                if len(batch) >= self.batch_size:
+                    if not self.settings.dry_run:
+                        db.DbSNP.insert_many(batch).execute()
+                    batch = []
+            db.database.commit()
```

Comment:
I guess there's an implicit commit once the `with ...atomic()` exits, but in that case why commit just before the last batch?

Reply:
dbsnp removed